### DISCUSSION TITLE
Is it possible to compile and run a function without a creating classes.

### DISCUSSION THREAD

**Discussion by eantar** [twinbasic/twinbasic#312] [General]
Forgive my ignorance in this if the questions doesn't make sense, as I'm a Access/VBA programmer with not a lot of training in the C world.

Is it possible to create a DLL with functions that are not associate with a class? Let's say I just wanted a library of compiled functions that could be called from inside my Access project. Is there a way to just...

MyResult = MyCompiledFunction()

Do that, without having to instantiate a class? So in the video example provided, it would be iResult = MultiplyByTen(123).

Thanks.

**Comment by WaynePhillipsEA** [comment]
Not at the moment.  VB6 had the same restriction, but support for this is planned in #22 by allowing for DLL exports.

**Comment by WaynePhillipsEA** [comment]
An alternative is to use predeclared classes.  By defining a class with the [PredeclaredId] attribute, the referencing VBA project creates an implicit global instance of the class in a global variable with the same name as the class.  (The semantics are more accurately the same as if defining a public global variable with the As New syntax).

So you can then call members of the class by using the syntax MyClass.MultiplyByTen() without explicitly instantiating MyClass.

**Reply by eantar** [reply]
Wayne! Thanks for getting back to me so quickly. I get you. So you wouldn't need to instantiate the class variable, but you would still need to preface the function / method with the class, ie MyClass.MultiplyByTen(). But NOT the ability to simply call MultiplyByTne(123), correct?

**Reply by WaynePhillipsEA** [reply]
Yes, that's right.   Technically it's also possible without the prefix, but it requires support for the equivalent of the VB6 VB_GlobalNameSpace attribute, which isn't yet supported by twinBASIC.

**Reply by WaynePhillipsEA** [reply]
linking issue #316 for the appobject / VB_GlobalNameSpace support

**Comment by eantar** [comment]
Awesome. Thanks.

Eddie

**Reply by WaynePhillipsEA** [reply]
twinBASIC now includes a [AppObject] attribute that you can set on classes to allow unqualified access to members of (implicitly) predeclared classes.  See https://github.com/WaynePhillipsEA/twinbasic/issues/316
